new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/TestUtils.java

```diff
@@ -88,6 +89,66 @@ public static boolean callWebAppAndCheckForServerNameInResponse(
     return false;
   }
 
+  /**
+   * Call the curl command and check the managed servers can see each other.
+   *
+   * @param curlCmd curl command to call the clusterview app
+   * @param managedServerNames managed server names that should be see other
+   * @param maxIterations max iterations to call the curl command
+   * @return true if the managed servers can see each other, false otherwise
+   */
+  public static boolean verifyClusterMemberCommunication(
+      String curlCmd,
+      List<String> managedServerNames,
+      int maxIterations) {
+
+    // first map all server names with false
+    HashMap<String, Boolean> managedServers = new HashMap<>();
+    managedServerNames.forEach(managedServerName
+        -> managedServers.put(managedServerName, false)
+    );
+
+    logger.info("Calling clusterview at most {0} times using command: {1}", maxIterations, curlCmd);
+
+    // check the response contains managed server name
+    ExecResult result = null;
+    for (int i = 0; i < maxIterations; i++) {
+      if (managedServers.containsValue(false)) {
+        try {
+          TimeUnit.MILLISECONDS.sleep(100);
+          result = ExecCommand.exec(curlCmd, true);
+          String response = result.stdout().trim();
+          for (var entry : managedServers.entrySet()) {
+            if (response.contains("ServerName:" + entry.getKey())) {
+              boolean bound = true;
+              for (String managedServerName : managedServerNames) {
+                bound = bound && response.contains("Bound:" + managedServerName);
+              }
+              if (bound) {
+                managedServers.put(entry.getKey(), true);
+              }
+            }
+          }
+        } catch (IOException | InterruptedException e) {
+          logger.info(e.toString());
+          return false;
+        }
+      } else {
+        return true;
+      }
+    }
+    // after the max iterations, if hit here, one or more servers cannot see other
+    managedServers.forEach((key, value) -> {
+      if (value) {
+        logger.info("The server {0} can see other cluster members", key);
+      } else {
+        logger.info("The server {0} cannot see other cluster members", key);
```

Comment:
Another possible reason is that the response contained unexpected content (did not contain ServerName/Bound).  It'd be helpful to accumulate an 'error' string that contains 'unexpected' responses (e.g. does not contain at least ServerName of one of the servers), and dump the String here on a failure.

Reply:
And yet another is that all managedServers didn't respond.   Or an 'extra' managed server responded.

Reply:
> 
> 
> Another possible reason is that the response contained unexpected content (did not contain ServerName/Bound). It'd be helpful to accumulate an 'error' string that contains 'unexpected' responses (e.g. does not contain at least ServerName of one of the servers), and dump the String here on a failure.

Its already there, when http request is made in a loop it will print this until all expected servers are bound.

<!DOCTYPE html>
<html>
<head>
<title>ClusterViewServlet</title>
</head>
<body>
ServerName:ms-2
Alive:3
Health:0
Members:ms-3,ms-1,ms-2
Unable to resolve 'ms-3'. Resolved ''
Unable to resolve 'ms-1'. Resolved ''
Bound:ms-2
</body>
</html>

Reply:
- I don't see how it's already there? I see line 120 `String response = result.stdout().trim();` but no corresponding print for the response if the response doesn't contain ServerName:.
- Regardless, the error message at line 145 itself is misleading if it hits the error cases I mentioned above (e.g. an error case that isn't due to 'unbound').

Reply:
The response from the server is always printed, that happens when the curl command is executed. line # 119
So it unnecessary to accumulate the unexpected response and print it again.

I can change the error message in line # 145 to include the case for unbound.

Please see the https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/323/consoleFull

search for `Unable to resolve 'ms-1'. Resolved ''`